src/Fmt_ast.ml

```diff
@@ -845,41 +845,52 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
   | Ppat_record (flds, closed_flag) ->
       let fmt_field (lid1, pat) =
         let {ppat_desc; ppat_loc; ppat_attributes} = pat in
+        let maybe_cmts = Option.value_map ~default:Fn.id ~f:(Cmts.fmt c) in
+        let fmt_field ?field_loc ?cnstr_loc ?ident_loc ?typ ?pat ~ctx c =
+          let fmt_type ?(parens = false) t =
+            fmt_if Poly.(c.conf.field_space = `Loose) " "
+            $ str ": "
+            $ fmt_core_type c (sub_typ ~ctx t)
+            $ fmt_if parens ")"
+          in
+          let fmt_pat ?(parens = false) ~ctx p =
+            fmt "=@;<1 2>" $ fmt_if parens "("
+            $ cbox 0 (fmt_pattern c (sub_pat ~ctx p))
+          in
+          maybe_cmts field_loc @@ maybe_cmts cnstr_loc
+          @@ cbox 0
+               ( maybe_cmts ident_loc @@ fmt_longident_loc c lid1
+               $
+               match (typ, pat) with
+               | Some t, Some p -> (
+                 match Source.typed_pattern t p with
+                 | `Type_first -> fmt_type t $ fmt_pat ~ctx p
+                 | `Pat_first ->
+                     fmt_pat ~ctx ~parens:true p $ fmt_type ~parens:true t )
+               | _ -> opt typ fmt_type $ opt pat (fmt_pat ~ctx) )
+        in
```

Comment:
This code is almost identical to the one in `Pexp_record`. I think factorization is required.

Reply:
I tried to generalize the function but since it operates on different types it is too much trouble, and I'm not sure it would be easier to read.
Or maybe there is a simplification I'm not seeing (feel free to push).

Reply:
I've made a PR to your PR: https://github.com/gpetiot/ocamlformat/pull/1